react/features/recording/components/LiveStream/GoogleSigninForm.native.js

```diff
@@ -0,0 +1,240 @@
+// @flow
+
+import React, { Component } from 'react';
+import { Text, View } from 'react-native';
+import { connect } from 'react-redux';
+import {
+    GoogleSignin
+} from 'react-native-google-signin';
+
+import { translate } from '../../../base/i18n';
+
+import {
+    GOOGLE_API_STATES,
+    GOOGLE_SCOPE_YOUTUBE,
+    GoogleSignInButton,
+    setGoogleAPIState
+} from '../../../google-api';
+
+import styles from './styles';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+/**
+ * Prop type of the component {@code GoogleSigninForm}.
+ */
+type Props = {
+
+    /**
+     * The ID for the Google client application used for making stream key
+     * related requests.
+     */
+    clientId: string,
+
+    /**
+     * The Redux dispatch Function.
+     */
+    dispatch: Function,
+
+    /**
+     * The ID for the Google client application used for making stream key
+     * related requests on iOS.
+     */
+    iOSClientId: string,
+
+    /**
+     * A callback to be invoked when an authenticated user changes, so
+     * then we can get (or clear) the YouTube stream key.
+     */
+    onUserChanged: Function,
+
+    /**
+     * Function to be used to translate i18n labels.
+     */
+    t: Function
+};
+
+/**
+ * State type of the component {@code GoogleSigninForm}.
+ */
+type State = {
+
+    /**
+     * The current state of the google api lib.
+     */
+    apiState: number,
+
+    /**
+     * The recent response that we received from Google.
+     */
+    googleResponse: ?Object
+};
+
+/**
+ * Class to render a google sign in form, or a google stream picker dialog.
+ *
+ * @extends Component
+ */
+class GoogleSigninForm extends Component<Props, State> {
+    /**
+     * Instantiates a new {@code GoogleSigninForm} component.
+     *
+     * @inheritdoc
+     */
+    constructor() {
+        super();
+
+        this.state = {
+            apiState: GOOGLE_API_STATES.NEEDS_LOADING,
+            googleResponse: undefined
+        };
+
+        this._onGoogleButtonPress = this._onGoogleButtonPress.bind(this);
+    }
+
+    /**
+     * Implements React's Component.componentDidMount.
+     *
+     * @inheritdoc
+     */
+    componentDidMount() {
+        GoogleSignin.hasPlayServices().then(() => {
+            if (this.props.clientId) {
+                GoogleSignin.configure({
+                    iosClientId: this.props.iOSClientId,
+                    offlineAccess: false,
+                    scopes: [ GOOGLE_SCOPE_YOUTUBE ],
+                    webClientId: this.props.clientId
```

Comment:
Why the need for both iosClientId and webClientId?

Reply:
Android uses the native Google auth routine of the platform, that's why it uses the same ID used on web and every other places. iOS is different (as always...) and it needs a different ID. See here: https://github.com/react-native-community/react-native-google-signin#configureconfiguration